libs/ngrx-toolkit/src/lib/with-mutations.ts

```diff
@@ -0,0 +1,169 @@
+import { computed, Signal } from '@angular/core';
+import {
+  EmptyFeatureResult,
+  signalStoreFeature,
+  SignalStoreFeature,
+  SignalStoreFeatureResult,
+  StateSignals,
+  StateSource,
+  withComputed,
+  withMethods,
+  WritableStateSource,
+} from '@ngrx/signals';
+
+export type Mutation<P, R> = {
+  (params: P): Promise<MutationResult<R>>;
+  status: Signal<MutationStatus>;
+  callCount: Signal<number>;
+  error: Signal<unknown>;
+};
+
+// NamedMutationMethods below will infer the actual parameter and return types
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type MutationsDictionary = Record<string, Mutation<any, any>>;
+
+export type MutationResult<T> =
+  | {
+      status: 'success';
+      value: T;
+    }
+  | {
+      status: 'error';
+      error: unknown;
+    }
+  | {
+      status: 'aborted';
+    };
+
+export type MutationStatus = 'idle' | 'processing' | 'error' | 'success';
+
+// withMethods uses Record<string, Function> internally
+// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+export type MethodsDictionary = Record<string, Function>;
+
+type NamedMutationProps<T extends MutationsDictionary> = {
+  [Prop in keyof T as `${Prop & string}Processing`]: Signal<boolean>;
```

Comment:
I would rename it to `IsProcessing`. `incrementerIsProcessing` might be easier to read than `incrementerProcessing`.